c7n/resources/s3.py

```diff
@@ -447,6 +447,46 @@ def __call__(self, b):
             return False
         return True
 
+@filters.register('missing-lifecycle-rule')
+class MissingLifecycleRuleFilter(Filter):
+    """Find buckets missing a set of named Lifecycle rules.
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-bucket-missing-lifecycle-rule
+                resource: s3
+                filters:
+                  - type: missing-lifecycle-rule
+                    rule_names:
+                      - AbortIncompleteMultipartUpload
+    """
+
+    schema = type_schema(
+        'missing-lifecycle-rule',
+        aliases=('missing-lifecycle',),
+        rule_names={'type': 'array', 'items': {'type': 'string'}})
+
+    permissions = ('s3:GetBucketLifecycle',)
+
+    def __call__(self, b):
+
+        lc = b.get('Lifecycle')
+        if lc is None:
+            return b
+
+        required = list(self.data.get('rule_names', []))
+        rules = lc.get('Rules', [])
+
+        for l in list(rules):
```

Comment:
the one letter variables here feel a bit obscure, l == rule? .. let's just use rule. also extraneous list( cast